File: Power of Two/Power_of_Two.py
# Runtime: 36 ms, faster than 100.00% of Python3 online submissions for Power of Two.
# Memory Usage: 13.2 MB, less than 6.20% of Python3 online submissions for Power of Two.
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n <= 0:
            return False
        else:
            return not (n - 1) & n

File: Power of Two/Power_of_Two.cpp
// 很明显的位运算的题
// 这个题目使用了 (n - 1) & n 的trick
// Runtime: 4 ms, faster than 100.00% of C++ online submissions for Power of Two.
// Memory Usage: 8.1 MB, less than 43.40% of C++ online submissions for Power of Two.
class Solution 
{
public:
    bool isPowerOfTwo(int n) 
    {
        if (n <= 0)
            return false;
        else
            return !((n - 1) & n);    
    }
};